hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBDefinitionFactory.java

```diff
@@ -98,6 +102,56 @@ public static DBDefinition getDefinition(Path dbPath,
     return getDefinition(dbName);
   }
 
+  public static List<CheckedSupplier<DBDefinition, Exception>> getFactories(
+      Class<?> clazz, String dbPathString, ConfigurationSource config) {
+    return Arrays.asList(
+        () -> {
+          Method factoryMethod = clazz.getDeclaredMethod("get");
+          factoryMethod.setAccessible(true);
+          return (DBDefinition) factoryMethod.invoke(clazz);
+        },
+        () -> {
+          Constructor<?> constructor = clazz.getDeclaredConstructor();
+          constructor.setAccessible(true);
+          return (DBDefinition) constructor.newInstance();
+        },
+        () -> {
+          Constructor<?> constructor = clazz.getDeclaredConstructor(String.class);
+          constructor.setAccessible(true);
+          return (DBDefinition) constructor.newInstance(dbPathString);
+        },
+        () -> {
+          Constructor<?> constructor = clazz.getDeclaredConstructor(String.class, ConfigurationSource.class);
+          constructor.setAccessible(true);
+          return (DBDefinition) constructor.newInstance(dbPathString, config);
+        }
```

Comment:
On second thought, we might want to reorder these, to ensure the constructor with more params is picked first (so that the config and DB path are not ignored).